pepper/exo_compute/decompose_bits.py

```diff
@@ -13,6 +13,8 @@
 
 # Assuming denominator is always 1
 number = int(fraction.split('%')[0])
+if number < 0:
+    number += 21888242871839275222246405745257275088548364400416034343698204186575808495617
```

Comment:
What if 

```
prime = 21888242871839275222246405745257275088548364400416034343698204186575808495617
number =  - prime - 1
```
Are we assuming this doesn't happen?

Because this would wind up being `number = -1` and still pass the assertion.

Reply:
> number =  - prime - 1

yeah, this is an issue. But hopefully, we can find a good way to deal with overflows. I would consider - prime -1 an overflow, exactly as any number bigger than prime.

Need to think about this

Reply:
Good point. We should be defensive and just take number % p here instead of adding it once in certain cases.